src/_igraph/graphobject.c

```diff
@@ -565,10 +565,21 @@ PyObject *igraphmodule_Graph_add_vertices(igraphmodule_GraphObject * self,
  */
 PyObject *igraphmodule_Graph_delete_vertices(igraphmodule_GraphObject * self,
                                              PyObject * args, PyObject * kwds) {
-  PyObject *list;
+  PyObject *list = 0;
```

Comment:
We need to do the same changes in `igraphmodule_Graph_delete_edges()` instead of doing it in the Python implementation of `delete_edges`; the reason is that if we do this in the Python layer, one could still call `GraphBase.delete_edges(graph, None)` to delete all edges.

(I know it's weird, but I have seen novice Python programmers to use the base class like this so we need to cover our bases).

Reply:
yep, I moved it to the C API now